service/awsconfig/v10/resource/cloudformation/update.go

```diff
@@ -112,13 +147,18 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 	// is different compared to updates. We can just process scaling any time. We
 	// cannot just process updates at any time and thus have to separate the
 	// management of both primitives. Note that updates also manage scaling at the
-	// same time for more efficiency.
+	// same time for more efficiency. Note that we have to preserve the master
+	// instance resource name when scaling worker nodes to prevent updating the
+	// master node. This is why we set the desired state of the master instance
+	// resource name to the current state below.
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the guest cluster main stack has to be scaled")
 
 		if shouldScale(currentStackState, desiredStackState) {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack has to be scaled")
 
+			desiredStackState.MasterInstanceResourceName = currentStackState.MasterInstanceResourceName
```

Comment:
Was master was rolled on scaling before?

Reply:
Yes, because the timestamp magic always generates a new name which makes CF roll the node even though we only want to scale workers. So we have to preserve the name in this case. Other updates are also covered above.

Reply:
Ack.